### ISSUE TITLE
CUSPARSE CI failures

### ISSUE THREAD

**Issue by maleadt** [JuliaGPU/CUDA.jl#1692]
Lately, there's been recurring CI failures in the CUSPARSE test suite, e.g., https://buildkite.com/julialang/cuda-dot-jl/builds/3374#0184cd77-ca80-4cb0-bd58-4c0539dc4aaf:

```
cusparse: Error During Test at /var/lib/buildkite-agent/builds/gpuci-14/julialang/cuda-dot-jl/test/cusparse.jl:157
  Test threw exception
  Expression: sparse(rows, cols, vals, fmt = :coo) isa CuSparseMatrixCOO{Float32}
  CUSPARSEError: invalid value (code 3, CUSPARSE_STATUS_INVALID_VALUE)
  Stacktrace:
    [1] throw_api_error(res::CUDA.CUSPARSE.cusparseStatus_t)
      @ CUDA.CUSPARSE /var/lib/buildkite-agent/builds/gpuci-14/julialang/cuda-dot-jl/lib/cusparse/libcusparse.jl:11
    [2] macro expansion
      @ /var/lib/buildkite-agent/builds/gpuci-14/julialang/cuda-dot-jl/lib/cusparse/libcusparse.jl:24 [inlined]
    [3] cusparseXcoosortByRow(handle::Ptr{CUDA.CUSPARSE.cusparseContext}, m::Int64, n::Int64, nnz::Int32, cooRowsA::CuArray{Int32, 1, CUDA.Mem.DeviceBuffer}, cooColsA::CuArray{Int32, 1, CUDA.Mem.DeviceBuffer}, P::CuArray{Int32, 1, CUDA.Mem.DeviceBuffer}, pBuffer::CuArray{UInt8, 1, CUDA.Mem.DeviceBuffer})
      @ CUDA.CUSPARSE /var/lib/buildkite-agent/builds/gpuci-14/julialang/cuda-dot-jl/lib/utils/call.jl:26
    [4] #2287
      @ /var/lib/buildkite-agent/builds/gpuci-14/julialang/cuda-dot-jl/lib/cusparse/conversions.jl:114 [inlined]
    [5] with_workspace(f::CUDA.CUSPARSE.var"#2287#2289"{CuSparseMatrixCOO{Float32, Int32}, Char, CuArray{Int32, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int32, 1, CUDA.Mem.DeviceBuffer}, CuArray{Int32, 1, CUDA.Mem.DeviceBuffer}, Int64, Int64}, eltyp::Type{UInt8}, size::CUDA.CUSPARSE.var"#bufferSize#2288"{CuSparseMatrixCOO{Float32, Int32}, Int64, Int64}, fallback::Nothing; keep::Bool)
      @ CUDA.APIUtils /var/lib/buildkite-agent/builds/gpuci-14/julialang/cuda-dot-jl/lib/utils/call.jl:77
    [6] with_workspace(f::Function, eltyp::Type{UInt8}, size::CUDA.CUSPARSE.var"#bufferSize#2288"{CuSparseMatrixCOO{Float32, Int32}, Int64, Int64}, fallback::Nothing)
      @ CUDA.APIUtils /var/lib/buildkite-agent/builds/gpuci-14/julialang/cuda-dot-jl/lib/utils/call.jl:58
    [7] #with_workspace#1
      @ /var/lib/buildkite-agent/builds/gpuci-14/julialang/cuda-dot-jl/lib/utils/call.jl:53 [inlined]
    [8] with_workspace (repeats 2 times)
      @ /var/lib/buildkite-agent/builds/gpuci-14/julialang/cuda-dot-jl/lib/utils/call.jl:53 [inlined]
    [9] sort_coo(A::CuSparseMatrixCOO{Float32, Int32}, type::Char)
      @ CUDA.CUSPARSE /var/lib/buildkite-agent/builds/gpuci-14/julialang/cuda-dot-jl/lib/cusparse/conversions.jl:113
   [10] #sparse#2280
      @ /var/lib/buildkite-agent/builds/gpuci-14/julialang/cuda-dot-jl/lib/cusparse/conversions.jl:42 [inlined]
   [11] sparse(I::CuArray{Int32, 1, CUDA.Mem.DeviceBuffer}, J::CuArray{Int32, 1, CUDA.Mem.DeviceBuffer}, V::CuArray{Float32, 1, CUDA.Mem.DeviceBuffer}; kws::Base.Iterators.Pairs{Symbol, Symbol, Tuple{Symbol}, NamedTuple{(:fmt,), Tuple{Symbol}}})
      @ CUDA.CUSPARSE /var/lib/buildkite-agent/builds/gpuci-14/julialang/cuda-dot-jl/lib/cusparse/conversions.jl:26
   [12] macro expansion
      @ /var/lib/buildkite-agent/builds/gpuci-14/julialang/cuda-dot-jl/test/cusparse.jl:157 [inlined]
   [13] macro expansion
      @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [14] macro expansion
      @ /var/lib/buildkite-agent/builds/gpuci-14/julialang/cuda-dot-jl/test/cusparse.jl:154 [inlined]
   [15] macro expansion
      @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [16] top-level scope
      @ /var/lib/buildkite-agent/builds/gpuci-14/julialang/cuda-dot-jl/test/cusparse.jl:111
```

Recent changes to CUSPARSE have been by @amontoison and @albertomercurio; any idea?

**Comment by amontoison** [comment]
We initialize the vectors with `undef` values for this [test](https://github.com/JuliaGPU/CUDA.jl/blob/master/test/cusparse.jl#L153-L159), I think that in some cases the vectors have at least one `NaN` and the sort routine is unable to sort the COO matrix by row.
I will update a PR to fix the test.

**Comment by amontoison** [comment]
I sent a message about this issue last month to CUDA developers and they didn't find any issue from their side.

Off-topic: It seems that they released the version 12.0 of CUDA toolkit. It's a great news because it should fix a lot of issue with CUSPARSE routines.
